src/test/kotlin/org/rust/ide/hints/RsInlayParameterHintsProviderTest.kt

```diff
@@ -275,6 +275,100 @@ class RsInlayParameterHintsProviderTest : RsTestBase() {
         check(inlays.size == 1)
     }
 
+    fun `test struct literal first`() = checkByText<RsStructLiteral>("""
+        struct S {
+            x: i32,
+            y: f64,
+        }
+        fn main() {
+            fn foo() -> S {
+                S {
+                    x:/*caret*/
+                }
+            }
+        }
+    """, " i32", 0)
+
+    fun `test struct literal second`() = checkByText<RsStructLiteral>("""
+        struct S {
+            x: i32,
+            y: f64,
+        }
+        fn main() {
+            fn foo() -> S {
+                S {
+                    x: 5,
+                    y:/*caret*/
+                }
+            }
+        }
+    """, " f64", 1)
+
+    fun `test generic struct literal`() = checkByText<RsStructLiteral>("""
+        pub trait Trait {}
+        struct S<T>;
+        struct B<T> {
+            x: i32,
+            y: T,
+            z: String,
+        }
+        fn main() {
+            fn foo<T: Trait>() -> B<S<T>> {
+                B {
+                    x: 3,
+                    y:/*caret*/
+                }
+            }
+        }
+    """, " S<T>", 1)
+
+    fun `test generic struct literal 2`() = checkByText<RsStructLiteral>("""
+        struct S<T>;
+        struct B<T> {
+            x: i32,
+            y: T,
+            z: String,
+        }
+        fn main() {
+            fn foo() -> B<S<i64>> {
+                B {
+                    x: 3,
+                    y:/*caret*/
+                }
+            }
+        }
+    """, " S<i64>", 1)
+
+    fun `test struct literal with tuple`() = checkByText<RsStructLiteral>("""
+        struct S<T>;
+        struct B<T> {
+            x: i32,
+            y: (i32, T),
+        }
+        fn main() {
+            fn foo() -> B<S<i64>> {
+                B {
+                    x: 3,
+                    y:/*caret*/
+                }
+            }
+        }
+    """, " (i32, S<i64>)", 1)
+
+
+    fun `test struct literal with fun`() = checkByText<RsStructLiteral>("""
+        struct S {
+            data: fn(i32) -> (fn(i32) -> i32),
+        }
+
+        fn foo(x: i32) -> S {
```

Comment:
Actually, I think, that this test adds no value, it should be already tested in type renderer, not here. I mean, that coverage for changed logic will not change, if this test would be deleted.